.github/workflows/book.yml

```diff
@@ -0,0 +1,94 @@
+# Performs checks, builds and deploys the mdbook
+
+name: book
+
+on:
+  push:
+    branches: [main, next]
+  pull_request:
+    paths:
+      - 'docs/src/SUMMARY.md'
+      - 'docs/src/IMPORTED.md'
+
+jobs:
+  check-files:
+    name: check interdependent files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check if SUMMARY.md and IMPORTED.md are modified together
+        run: |
+          if [ "${{ github.event_name }}" == "pull_request" ]; then
+            BASE_SHA=${{ github.event.pull_request.base.sha }}
+            HEAD_SHA=${{ github.event.pull_request.head.sha }}
+          else
+            # For push events, compare with the previous commit
+            BASE_SHA=${{ github.event.before }}
+            HEAD_SHA=${{ github.event.after }}
+          fi
+
+          # Check which files were modified
+          SUMMARY_CHANGED=$(git diff --name-only $BASE_SHA $HEAD_SHA | grep -q "src/docs/SUMMARY.md" && echo "true" || echo "false")
+          IMPORTED_CHANGED=$(git diff --name-only $BASE_SHA $HEAD_SHA | grep -q "src/docs/IMPORTED.md" && echo "true" || echo "false")
+
+          # If one file changed and the other didn't, fail the check
+          if [[ "$SUMMARY_CHANGED" == "true" && "$IMPORTED_CHANGED" == "false" ]]; then
+            echo "Error: src/docs/SUMMARY.md was modified but src/docs/IMPORTED.md was not. Please update both files together to make sure the local book and the Miden book are in sync."
+            exit 1
+          fi
+
+          if [[ "$SUMMARY_CHANGED" == "false" && "$IMPORTED_CHANGED" == "true" ]]; then
+            echo "Error: src/docs/IMPORTED.md was modified but src/docs/SUMMARY.md was not. Please update both files together to make sure the local book and the Miden book are in sync."
+            exit 1
+          fi
+
+          echo "Both files were either modified together or not modified at all. Check passes!"
+
+  deploy:
+    name: deploy mdbook
+    runs-on: ubuntu-latest
+    needs: check-files
+    # Only run deployment on push to main/next branches, not on PRs
+    if: github.event_name == 'push' || github.event_name == 'workflow_run'
```

Comment:
This means pushes to `main` and `next` will overwrite each other. Probably we would want this to only happen on `next` so that they're always the latest dev docs.